src/plugins/desktop/testmod_desktop.c

```diff
@@ -0,0 +1,72 @@
+/**
+ * @file
+ *
+ * @brief
+ *
+ * @copyright BSD License (see LICENSE.md or https://www.libelektra.org)
+ */
+
+#ifdef HAVE_KDBCONFIG_H
+#include "kdbconfig.h"
+#endif
+
+#include <stdio.h>
+#ifdef HAVE_STDLIB_H
+#include <stdlib.h>
+#endif
+#ifdef HAVE_STRING_H
+#include <string.h>
+#endif
+
+#include <tests_plugin.h>
+
+const char DE_ENV[] = "XDG_CURRENT_DESKTOP";
+const char TEST_DE[] = "this_is_a_test_de";
+const char NEW_DE[] = "this_is_another_test_de";
+
+void test_desktop (void)
+{
+
+	setenv (DE_ENV, TEST_DE, 1);
+
+	Key * parentKey = keyNew ("user:/tests/desktop", KEY_END);
+	KeySet * keys = ksNew (0, KS_END);
+	KeySet * conf = 0;
+
+	PLUGIN_OPEN ("desktop")
+	succeed_if (plugin->kdbGet (plugin, keys, parentKey) == 1, "could not call kdbGet")
```

Comment:
Better to also use macros like function calls:

```suggestion
	PLUGIN_OPEN ("desktop");
	succeed_if (plugin->kdbGet (plugin, keys, parentKey) == 1, "could not call kdbGet");
```